core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { MapLayerSettings } from "@bentley/imodeljs-common";
+import { expect } from "chai";
+
+import {
+  ArcGISMapLayerImageryProvider,
+  BingMapsImageryLayerProvider,
+  internalMapLayerImageryFormats,
+  MapBoxLayerImageryProvider,
+  MapLayerFormatRegistry,
+  TileUrlImageryProvider,
+  WmsMapLayerImageryProvider,
+  WmtsMapLayerImageryProvider,
+} from "../../../tile/internal";
+
+const getSampleLayerSettings = ((formatId: string) => {
+  return MapLayerSettings.fromJSON({
+    formatId, url: "https://localhost/service", name: `Test ${formatId}`,
+  });
+});
+
+describe("MapLayerImageryFormats", () => {
+  it("should create proper provider", () => {
+    const registry = new MapLayerFormatRegistry({});
+    internalMapLayerImageryFormats.forEach((imageryFormat) => {
+
+      const layerSettings = getSampleLayerSettings(imageryFormat.formatId);
+      expect(layerSettings).to.not.undefined;
+      if (!layerSettings)
+        return;
+      const provider = registry.createImageryProvider(layerSettings);
+
+      expect(provider).to.not.undefined;
+      if (!layerSettings)
```

Comment:
`/s/layerSettings/provider` ?

Reply:
Actually no check is needed at all after creating the provider, I removed it.